app/scripts/models/flow.js

```diff
@@ -0,0 +1,107 @@
+ * need to submit as part of our metrics when we perform actions
+ * during the flow.
+ *
+ * Try to fetch the flowId and flowBegin from the resume token.
+ * If unavailable there, fetch from data attributes in the DOM.
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var AuthErrors = require('lib/auth-errors');
+  var Backbone = require('backbone');
+  var Cocktail = require('cocktail');
+  var ErrorUtils = require('lib/error-utils');
+  var ResumeTokenMixin = require('models/mixins/resume-token');
+  var SearchParamMixin = require('models/mixins/search-param');
+  var vat = require('lib/vat');
+
+  var Model = Backbone.Model.extend({
+    initialize: function (options) {
+      options = options || {};
+
+      this.sentryMetrics = options.sentryMetrics;
+      this.window = options.window || window;
+
+      this.fetch();
+    },
+
+    defaults: {
+      flowBegin: null,
+      flowId: null
+    },
+
+    fetch: function () {
+
+      var error;
+      var $body = this.window['$']('body');
+
+      this.populateFromStringifiedResumeToken(this.getSearchParam('resume'));
+
+      // We should either get both fields from the resume token, or neither.
+      // Getting one without the other is an error.
+
+      if (this.get('flowId')) {
+        if (! this.get('flowBegin')) {
+          error = AuthErrors.toMissingResumeTokenPropertyError('flowBegin');
+          ErrorUtils.captureError(error, this.sentryMetrics);
+        }
+      } else if (this.get('flowBegin')) {
+        error = AuthErrors.toMissingResumeTokenPropertyError('flowId');
+        ErrorUtils.captureError(error, this.sentryMetrics);
+      } else {
+        // No flow information in the resume token,
+        // read it afresh from data attributes in the DOM
+
+        var flowId = $body.data('flowId');
+        if (! flowId) {
+          error = AuthErrors.toMissingDataAttributeError('flowId');
+          ErrorUtils.captureError(error, this.sentryMetrics);
+        } else {
+          try {
+            flowId = this.resumeTokenSchema.flowId.validate(flowId);
+            this.set('flowId', flowId);
+          } catch (err) {
+            error = AuthErrors.toInvalidDataAttributeError('flowId');
+            ErrorUtils.captureError(error, this.sentryMetrics);
+          }
+        }
+
+        var flowBegin = $body.data('flowBegin');
+        if (! flowBegin) {
+          error = AuthErrors.toMissingDataAttributeError('flowBegin');
+          ErrorUtils.captureError(error, this.sentryMetrics);
+        } else {
+          try {
+            flowBegin = this.resumeTokenSchema.flowBegin.validate(flowBegin);
+            this.set('flowBegin', flowBegin|0);
+          } catch (err) {
+            error = AuthErrors.toInvalidDataAttributeError('flowBegin');
+            ErrorUtils.captureError(error, this.sentryMetrics);
+          }
+        }
+      }
+    },
+
+    resumeTokenFields: ['flowId', 'flowBegin'],
+
+    resumeTokenSchema: {
```

Comment:
An idea that I had takes a page from your book @philbooth - instead of unnaturally placing `resumeTokenFields` and `resumeTokenSchema` directly on the consuming module, ResumeTokenMixin returns a function, call the function with the config (which would include `resumeTokenFields` and `resumeTokenSchema`), and the function would return the actual mixin:

```js
Cocktail.mixin(
  Model,
  ResumeTokenMixin({
    schema: schema,
    fields: ['flowId', 'flowBegin']
  }),
  SeachParamMixin
);
```